src/main/java/api/rest/v2/samples/SamplesControllerV2.java

```diff
@@ -0,0 +1,126 @@
+package api.rest.v2.samples;
+
+
+import application.ApplicationException;
+import application.SampleService;
+import domain.InvalidDomainException;
+import domain.notification.INotificationRepository;
+import domain.sample.SampleEventDatasource;
+import domain.sample.SampleEventStore;
+import domain.sample.SampleRepository;
+import domain.sample.Status;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Error;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.PathVariable;
+import io.micronaut.http.annotation.Put;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import life.qbic.auth.Authentication;
+
+@Requires(beans = Authentication.class)
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Controller("/v2/samples")
+public class SamplesControllerV2 {
+
+  SampleService sampleService;
+
+  @Inject
+  public SamplesControllerV2(SampleEventDatasource sampleEventDatasource, INotificationRepository notificationRepository) {
+    this.sampleService = new SampleService(
+        new SampleRepository(new SampleEventStore(sampleEventDatasource)), notificationRepository);
+  }
+
+  @Put(uri = "/{sampleCode}/status")
+  @RolesAllowed("WRITER")
+  public HttpResponse<?> moveSampleToStatus(@PathVariable String sampleCode,
+      @Body StatusChangeRequest statusChangeRequest) {
+    String validFrom = statusChangeRequest.validSince;
+    String requestedStatus = statusChangeRequest.status;
+    if (SampleStatusDto.METADATA_REGISTERED.name().equals(requestedStatus)) {
+      sampleService.registerMetadata(sampleCode, validFrom);
+
+    } else if (SampleStatusDto.SAMPLE_RECEIVED.name().equals(requestedStatus)) {
+      sampleService.receiveSample(sampleCode, validFrom);
+
+    } else if (SampleStatusDto.SAMPLE_QC_FAIL.name().equals(requestedStatus)) {
+      sampleService.failQualityControl(sampleCode, validFrom);
+
+    } else if (SampleStatusDto.SAMPLE_QC_PASS.name().equals(requestedStatus)) {
+      sampleService.passQualityControl(sampleCode, validFrom);
+
+    } else if (SampleStatusDto.LIBRARY_PREP_FINISHED.name().equals(requestedStatus)) {
+      sampleService.prepareLibrary(sampleCode, validFrom);
+
+    } else if (SampleStatusDto.DATA_AVAILABLE.name().equals(requestedStatus)) {
+      sampleService.provideData(sampleCode, validFrom);
```

Comment:
```suggestion
    String validSince = statusChangeRequest.validSince;
    String requestedStatus = statusChangeRequest.status;
    if (SampleStatusDto.METADATA_REGISTERED.name().equals(requestedStatus)) {
      sampleService.registerMetadata(sampleCode, validSince);

    } else if (SampleStatusDto.SAMPLE_RECEIVED.name().equals(requestedStatus)) {
      sampleService.receiveSample(sampleCode, validSince);

    } else if (SampleStatusDto.SAMPLE_QC_FAIL.name().equals(requestedStatus)) {
      sampleService.failQualityControl(sampleCode, validSince);

    } else if (SampleStatusDto.SAMPLE_QC_PASS.name().equals(requestedStatus)) {
      sampleService.passQualityControl(sampleCode, validSince);

    } else if (SampleStatusDto.LIBRARY_PREP_FINISHED.name().equals(requestedStatus)) {
      sampleService.prepareLibrary(sampleCode, validSince);

    } else if (SampleStatusDto.DATA_AVAILABLE.name().equals(requestedStatus)) {
      sampleService.provideData(sampleCode, validSince);
```
Thank you for this change!